src/libexpr/lexer.l

```diff
@@ -302,8 +308,25 @@ or          { return OR_KW; }
 {URI}       { yylval->uri = {yytext, (size_t) yyleng}; return URI; }
 
 [ \t\r\n]+    /* eat up whitespace */
-\#[^\r\n]*    /* single-line comments */
-\/\*([^*]|\*+[^*/])*\*+\/  /* long comments */
+\#[^\r\n]*   {
+  // single-line comments
+  data->comments.emplace_back(Comment {
+    .kind = Comment::CK_Single,
+    .content = std::string(yytext, (size_t) yyleng),
+    .start = CUR_POS,
+    .end = END_POS
+  });
+}
+
+\/\*([^*]|\*+[^*/])*\*+\/  {
```

Comment:
doc-comments per rfc145 start with /** */, but this is nice though for testing against nixpkgs.lib